src/Symfony/Component/Validator/Constraints/Slug.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Validates that a value is a valid slug.
+ *
+ * @author Raffaele Carelle <raffaele.carelle@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Slug extends Constraint
+{
+    public const NOT_SLUG_ERROR = '14e6df1e-c8ab-4395-b6ce-04b132a3765e';
+
+    public string $message = 'This value is not a valid slug.';
```

Comment:
This message needs to be added in translation files (so that the translators from the community can provide translations for it in core, like other default messages of constraints)

Reply:
this has already been done in #59383